ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1026,6 +1026,17 @@ func (r *RayClusterReconciler) reconcileAutoscalerServiceAccount(instance *rayio
 			return err
 		}
 
+		// If users specify ServiceAccountName for the head Pod, they need to create a ServiceAccount themselves.
+		// However, if KubeRay creates a ServiceAccount for users, the autoscaler may encounter permission issues during
+		// zero-downtime rolling updates when RayService is performed. See https://github.com/ray-project/kuberay/issues/1123
+		// for more details.
+		if instance.Spec.HeadGroupSpec.Template.Spec.ServiceAccountName == namespacedName.Name {
```

Comment:
I agree that using `instance.Spec.HeadGroupSpec.Template.Spec.ServiceAccountName != ""` is more straightforward for the comment "if users have specified a ServiceAccountName for the head Pod".

However, if the logic in `GetHeadGroupServiceAccountName` changes, it may be difficult for developers to be aware that they also need to update the code logic here when using `... != ""`. That's why I decide to use `... != namespacedName.Name`.

Reply:
I see, thank you!